tests/generate.py

```diff
@@ -0,0 +1,531 @@
+"""
```

Comment:
Can you split the rename out into a separate commit? Reviewing this with the GitHub UI is too hard.